enumerable_method.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    each { |i| yield(i) }
+  end
+  # Self represent the current object, returns the elements of the array
+  # Yield asks for a block
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    # select: Returns a new array containing all elements of enum for which the given block returns a true value.
+    return to_enum unless block_given?
+
+    # new array
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  # my_all?
+
+  def my_all?
```

Comment:
**[REQUIRED]**
**1. When no block or argument is given returns true when none of the collection members are false or nil.**

Meaning: 
`[1, true, 'hi'].my_All?`should return true.
`[1, false, 'hi'].my_All?`should return false.

When a block and argument was not given. It's going to do something like this(You can use your own `each` here):
`self.my_each { |element| return false unless(element) }`
`return true`

**2. When a class is passed as an argument returns true if all of the collection is a member of such class.**
Meaning: 
`[1, 2, 3].my_All?(Integer)`should return true.
`[1, false, 'hi'].my_All?(Integer)`should return false.

When a class is passed as an argument, it's going to do something like this:

`if arg.class == Class`
      `self.my_each { |element| return false unless element.class == arg }`
`return true`

**3. When a Regex is passed as an argument returns true if all of the collection matches the Regex.**

Meaning: 
`%w[dog door rod blade].my_All?(/d/)` returns true
`%w[dog door rod blade].my_All?(/o/)` returns false

when a Regex is passed as an argument, it's going to do something like this:

`if arg.class == Regexp`
      `self.my_each { |element| return false if (element =~ arg).nil? }`
`return true`

**4. when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern**

Meaning: 
`[3,3,3].my_all?(3)` returns true
`[3,3,2].my_all?(3)` returns false